UI/window-basic-main.cpp

```diff
@@ -2065,6 +2065,45 @@ void OBSBasic::CreateFiltersWindow(obs_source_t *source)
 	filters->setAttribute(Qt::WA_DeleteOnClose, true);
 }
 
+static bool ShowSource(void *data, obs_hotkey_pair_id id, obs_hotkey_t *key,
+		bool pressed)
+{
+	UNUSED_PARAMETER(id);
+	UNUSED_PARAMETER(key);
+
+	obs_source_t *source = static_cast<obs_source_t*>(data);
+
+	if (!pressed || obs_source_enabled(source) || !source)
+		return false;
+
+	obs_source_set_enabled(source, true);
+	return true;
+}
+
+static bool HideSource(void *data, obs_hotkey_pair_id id, obs_hotkey_t *key,
+		bool pressed)
+{
+	UNUSED_PARAMETER(id);
+	UNUSED_PARAMETER(key);
+
+	obs_source_t *source = static_cast<obs_source_t*>(data);
+
+	if (!pressed || !obs_source_enabled(source) || !source)
```

Comment:
Is the null check (`!source`) required here?  Doesn't `obs_source_enabled(source)` already do this check?  See [libobs/obs-source.c#L3543](https://github.com/obsproject/obs-studio/blob/db06a1c12acb50422d61055fc7053e95048d6bb6/libobs/obs-source.c#L3543) and [libobs/obs-internal.h#L60](https://github.com/obsproject/obs-studio/blob/57121478acccd97b6c3cd909e0e8cbf7d7dda05a/libobs/obs-internal.h#L60) (lines 60-75).  If `source` would be evaluated as false, then `obs_object_valid` would return false, and then `obs_source_enabled` would return false, so `!obs_source_enabled` would be true if `!source` is true.  Similarly, if the source is valid and enabled, then `!obs_source_enabled` would be false and `!source` would also be false.  It seems to me that the `!source` condition would never be evaluated when `!obs_source_enabled` is true, and whenever it is evaluated, `!source` will always be false.

I realize that removing `!source` here may make it seem unbalanced compared to the one in `ShowSource` above.  Something to consider.